### PR TITLE
Fix/timerange picker analytics

### PR DESCRIPTION
**Pull request by alianza**
Add ability to select a full year

- Added absolute input with input years & input type
- Added case for absoluteInput type option
- Added case for absoluteInput to create UI components
- Added .tnic-input-large class for larger input fields

Fixes: https://linear.app/e-flux/issue/E-262#comment-0c5ee525

### CODE REVIEW

File: src/components/TimeRangePicker/index.tsx [hunk 1]
```diff
@@ -55,9 +55,22 @@ const defaultProps = {
       unit: "months",
       default: 3,
     },
+    {
```
Comment (kaareal) [inline src/components/TimeRangePicker/index.tsx:58]: E-flux should have their own sets of timeOptions, which doesnt have to same for other projects. 
So this should not be added here. 

Note the timeOptions can be overwritten as they are just defaultProps

**Review by kaareal — CHANGES_REQUESTED** [review]
This is too specific for the need for one project, this needs to be more general.

I would like this to be rolled into "type": "fixed" (or perhaps new type called "select")   

```
{
  type: "fixed",
  label: "Year",
  options: [{ value, label: "2020"  }, { value, "label": "2021" } .... ]
}
```

That would also allow us to handle handle 2021 q1, 2021 q2 etc 

And at same time pushing the logic of generate what values would be valid to the implementor.

File: src/components/TimeRangePicker/index.tsx (hunk 1 above)
Comment (alianza) [reply src/components/TimeRangePicker/index.tsx:58]: Ah I see they can be overwritten. But for the storyBook demo it it not useful to show all options initially?

**Comment by alianza** [comment]
> This is too specific for the need for one project, this needs to be more general.
> 
> I would like this to be rolled into "type": "fixed" (or perhaps new type called "select")
> 
> ```
> {
>   type: "fixed",
>   label: "Year",
>   options: [{ value, label: "2020"  }, { value, "label": "2021" } .... ]
> }
> ```
> 
> That would also allow us to handle handle 2021 q1, 2021 q2 etc
> 
> And at same time pushing the logic of generate what values would be valid to the implementor.

I've been working on this. I've been generalizing this as a new type called Select. I have this working with selectable year and month. Quarters I couldn't really get to work, generalizing all that in one type seems like more work than just separating them. Please review the current solution for selectable year or month, it works quite nicely!

File: src/components/TimeRangePicker/index.tsx (hunk 1 above)
Comment (kaareal) [reply src/components/TimeRangePicker/index.tsx:58]: I will fix, i will trim some of the default options while i am at, to force people to implement their defaults (if they want something special) As ideally there shouldn't be scroll.

Comment (alianza) [reply src/components/TimeRangePicker/index.tsx:58]: > I will fix, i will trim some of the default options while i am at, to force people to implement their defaults (if they want something special) As ideally there shouldn't be scroll.

Okay cool! Does anything still needs to be done before this is merged or is it ready?

**Pull request merged**